src/lib/perf_histograms/rpc.ml

```diff
@@ -7,9 +7,13 @@ let decorate_dispatch ~name (dispatch : ('q, 'r) Intf.dispatch) :
   let open Deferred.Or_error.Let_syntax in
   let start = Time.now () in
   let%map r = dispatch conn q in
-  Perf_histograms0.add_span
-    ~name:(sprintf "rpc_dispatch_%s" name)
-    (Time.diff (Time.now ()) start) ;
+  let span = Time.diff (Time.now ()) start in
+  Perf_histograms0.add_span ~name:(sprintf "rpc_dispatch_%s" name) span ;
+  Coda_metrics.(
+    Network.Rpc_histogram.observe Network.rpc_latency_ms_summary
+      (Time.Span.to_ms span)) ;
```

Comment:
Is there a reason you couldn't hook this into the `Perf_histograms` flow? Also we'd want to distinguish different metrics for the differently named rpcs